### ISSUE TITLE
[4.0.58] select2_ajax filter breaks "Remove filters" button (again)

### ISSUE THREAD

**Issue by FakeFakers** [Laravel-Backpack/CRUD#2645]
# Bug report
Previous issue - https://github.com/Laravel-Backpack/CRUD/issues/2135

### What I did
Same as previous issue. 

### What I expected to happen
The filters should be removed.

### What happened
The filter is removed from the interface, but the list return zero results.

When I run ```php artisan backpack:version``` the output is:
### PHP VERSION:
PHP 7.4.4 (cli) (built: Mar 20 2020 13:47:45) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.4, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
v6.18.3@4e48acfaba87f08320a2764d36c3b6a4a4112ccf

### BACKPACK VERSION:
4.0.58@03432680d8ed839bcbd99ce57aed61bcf7a1b695

**Comment by pxpm** [comment]
If you update your app to 4.1 there is already an reset button that will clear the localStorage from browser. In the meanwhile you need to do it manually from your developer tools interface in browser.

Let me know if that fixes the issue for you. 

Best

**Comment by tabacitu** [comment]
@pxpm is this a bug in 4.0? Or can we close it?

**Comment by pxpm** [comment]
@tabacitu i could not reproduce this in online demo. What version 4.0.** demo is running ?

I am going to close this, if you think this still an issue please re-open.

**Comment by tabacitu** [comment]
Ok so it looks like I've stumbled onto this problem myself.

Without a `select2_ajax` filter everything works fine. I click `remove filters` and all filters get removed, proper entries are shown:
![2020-09-02 07 54 59](https://user-images.githubusercontent.com/1032474/91978148-56461980-ed2c-11ea-866d-6b4ae122aa7f.gif)

However, if I add a `seelect2_ajax` filter and do the same thing (active multiple filters, then click the `remove filters` button),  it did NOT load all entries like you expected, it would say "No entries":
![2020-09-02 07 56 16](https://user-images.githubusercontent.com/1032474/91978489-da989c80-ed2c-11ea-8402-68130cacf3e3.gif)

It looks like what's happening is that the GET parameter of the `select2_ajax` filter ("user" in this case) keeps being added to the URL, even when it has no value:
![2020-09-02 08 01 00](https://user-images.githubusercontent.com/1032474/91978673-32370800-ed2d-11ea-9a53-1a6cdca8d792.gif)

Which means that my filter logic would basically try to just show stuff where user_id is NULL:
```php
        CRUD::filter('user')
            ->type('select2_ajax')
            ->label('User')
            ->values(backpack_url('net-promoter-answer/get-user-options'))
            ->whenActive(function ($value) {
                    $this->crud->addClause('where', 'user_id', $value);
            });
```

Now... this can easily be fixed in userland, by checking if the value is NULL:
```php
        CRUD::filter('user')
            ->type('select2_ajax')
            ->label('User')
            ->values(backpack_url('net-promoter-answer/get-user-options'))
            ->whenActive(function ($value) {
                if ($value != null) {
                    $this->crud->addClause('where', 'user_id', $value);    
                }
            });
```

But it's not intuitive, it's not the way the _other_ filters behave. So `select2_ajax` should behave this way either, it shouldn't need this check if the `$value` is null. If the `$value` is null the filter should not be active, it shouldn't even reach the closure.

So I think **we should prevent the `select2_ajax` filter from putting the GET parameter in the URL, when the value is NULL**.

**Comment by tabacitu** [comment]
Fixed by #3177 hopefully, so I'm going to close this.
